Formula/fypp.rb

```diff
@@ -0,0 +1,20 @@
+class Fypp < Formula
+  include Language::Python::Virtualenv
+
+  desc "Python powered Fortran preprocessor"
+  homepage "https://fypp.readthedocs.io/en/stable/"
+  url "https://github.com/aradi/fypp/archive/refs/tags/3.1.tar.gz"
+  sha256 "0f66e849869632978a8a0623ee510bb860a74004fdabfbfb542656c6c1a7eb5a"
+  license "BSD-2-Clause"
+  head "https://github.com/aradi/fypp.git", branch: "master"
+
+  depends_on "python@3.9"
+
+  def install
+    virtualenv_install_with_resources
+  end
+
+  test do
+    system "#{bin}/fypp", "--version"
```

Comment:
Can you propose a more substantial test than checking the version or calling for help? These tests rarely ever fail even for broken installations.

>We want tests that don’t require any user input and test the basic functionality of the application. For example `foo build-foo input.foo` is a good test and (despite their widespread use) `foo --version` and `foo --help` are bad tests. 

Ref. https://docs.brew.sh/Formula-Cookbook#add-a-test-to-the-formula